### DISCUSSION TITLE
Proposal: Compress Large Audio Files (< 25 MB) & Integrate GroqAI Transcription

### DISCUSSION THREAD

**Discussion by bekmuradov** [open-webui/open-webui#5236] [Ideas]
Hi,

I'd like to propose two enhancements for audio file handling in OpenWebUI:

1. **Compress large audio files**: Both OpenAI Whisper and GroqAI transcription endpoints have a file size limit of 25 MB. Currently, OpenWebUI doesn't validate file sizes during upload (in documents) or show progress while processing. It would be useful to automatically compress files larger than 25 MB to stay within this limit.

2. **Integrate GroqAI Transcription**: GroqAI transcription is faster than OpenAI Whisper, and adding it as an option could improve the overall user experience.

Why compress files? While users can compress files externally, having this feature built-in would streamline the workflow and keep everything in one place.

Looking forward to hearing your thoughts.

Thanks!  
Beck

**Comment by thiswillbeyourgithub** [comment]
I think limiting silence to 3s would help too. (Please dont' use pydub)


Edit: related: #5972
